aep/general/0231/aep.md.j2

```diff
@@ -1,5 +1,168 @@
 # Batch methods: Get
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/42) for more
-information.
+Some APIs need to allow users to get a specific set of resources at a
+consistent time point (e.g. using a read transaction). A batch get method
```

Comment:
If the transactional guarantee is dropped, how does "batchGet" differ from "List" with a filter that specifies the resource names/paths?

Reply:
That's a very fair point.  It's functionally equivalent (but may be able to be implemented much more efficiently, which is sometimes important - though I suppose one could write code that, at least under normal circumstances, detects when a filtered list request is equivalent to a batch get based on the filter provided).

We actually discussed this in general in the meeting today.  @toumorokoshi proposed that maybe we factor out the guidance about transactionality, explain that it's optional for each batch method, and then describe a pattern for partial failures to be used for non-transactional batch operations.

Reply:
If we merge this as-is and then remove this requirement, is that a breaking change?

Maybe it is if we add a requirement that transactional batch gets must document that they're transactional, because someone could go and implement one that doesn't document it?

Alternatively, maybe we could have different operation names for the non-transactional cases (e.g. MultiGetBooks vs. BatchGetBooks)?

Reply:
> If the transactional guarantee is dropped, how does "batchGet" differ from "List" with a filter that specifies the resource names/paths?

I find this very compelling. Given a not-necessarily-transactional List operation, BatchGet makes sense _only_ with such a guarantee.

Reply:
We've agreed to make transactionality optional -- controlled by an "asTransaction" parameter.